utils/nft-utils/src/main.rs

```diff
@@ -0,0 +1,217 @@
+    let transfer_message = RuntimeCall::<DefaultContext, MockDaSpec>::nft(CallMessage::<
+        DefaultContext,
+    >::TransferNft {
+        collection_address: collection_address.clone(),
+        token_id,
+        to: UserAddress::new(to),
+    });
+    Transaction::<DefaultContext>::new_signed_tx(
+        signer,
+        transfer_message.try_to_vec().unwrap(),
+        nonce,
+    )
+}
+
+#[tokio::main]
+async fn main() {
+    let creator_pk = DefaultPrivateKey::try_from(&PK1[..]).unwrap();
+    let owner_1_pk = DefaultPrivateKey::try_from(&PK2[..]).unwrap();
+    let owner_2_pk = DefaultPrivateKey::try_from(&PK3[..]).unwrap();
+
+    let client = SimpleClient::new("localhost", 12345).await.unwrap();
+
+    let mut nonce = 0;
+    let mut nft_id = 1;
+    let mut transactions = vec![];
+    transactions.push(create_collection(&creator_pk, nonce, COLLECTION_1).await);
+    nonce += 1;
+    transactions.push(create_collection(&creator_pk, nonce, COLLECTION_2).await);
+    nonce += 1;
+    transactions.push(create_collection(&creator_pk, nonce, COLLECTION_3).await);
+
+    client.send_transactions(transactions, None).await.unwrap();
+    // sleep is necessary because of how the sequencer currently works
+    // without the sleep, there is a concurrency issue and some transactions would be ignored
+    thread::sleep(Duration::from_millis(1000));
+
+    let mut transactions = vec![];
+    for _ in 0..15 {
+        nonce += 1;
+        transactions.push(
+            mint_nft(
+                &creator_pk,
+                nonce,
+                COLLECTION_1,
+                nft_id,
+                &owner_1_pk.default_address(),
+            )
+            .await,
+        );
+        nft_id += 1;
+    }
+
+    client.send_transactions(transactions, None).await.unwrap();
+    thread::sleep(Duration::from_millis(1000));
+
+    let mut transactions = vec![];
+    for _ in 0..5 {
+        nonce += 1;
+        transactions.push(
+            mint_nft(
+                &creator_pk,
+                nonce,
+                COLLECTION_1,
+                nft_id,
+                &owner_2_pk.default_address(),
+            )
+            .await,
+        );
+        nft_id += 1;
+    }
+
+    let mut nft_id = 1;
+    for _ in 0..20 {
+        nonce += 1;
+        transactions.push(
+            mint_nft(
+                &creator_pk,
+                nonce,
+                COLLECTION_2,
+                nft_id,
+                &owner_1_pk.default_address(),
+            )
+            .await,
+        );
+        nft_id += 1;
+    }
+
+    client.send_transactions(transactions, None).await.unwrap();
+    thread::sleep(Duration::from_millis(1000));
+
+    let mut transactions = vec![];
+
+    let mut owner_1_nonce = 0;
+    let mut nft_id = 1;
+    let collection_1_address = get_collection_address::<DefaultContext>(
+        COLLECTION_1,
+        creator_pk.default_address().as_ref(),
+    );
+
+    #[allow(clippy::explicit_counter_loop)]
```

Comment:
Can we use the loop counter instead of using separate `owner_1_nonce` and `nft_id` variables here? Will make clippy happy and remove a little code.